tdm_loader/tdm_loader.py

```diff
@@ -228,31 +228,27 @@ def channel_search(self, search_term):
         """
         search_term = str(search_term).upper().replace(" ", "")
 
-        ind_chg_ch = []
-        for j in range(len(self._xml_chgs)):
-            chs = self._channels_xml(j)
-
-            if search_term == "":
-                found_terms = [
-                    ch.findtext("name") for ch in chs if ch.findtext("name") is None
-                ]
-            else:
-                found_terms = [
-                    ch.findtext("name")
-                    for ch in chs
-                    if ch.findtext("name") is not None
-                    and ch.findtext("name")
-                    .upper()
-                    .replace(" ", "")
-                    .find(str(search_term))
-                    >= 0
-                ]
-
-            for name in found_terms:
-                i = [ch.findtext("name") for ch in chs].index(name)
-                ind_chg_ch.append((name, j, i))
-
-        return ind_chg_ch
+        matched_channels = []
+        channel_groups_cache = {}
+        channel_group_ids = {v: i for i, v in enumerate(x.get("id") for x in self._xml_chgs)}
+
+        for channel in self._root.findall(".//tdm_channel"):
+            if channel_name := channel.find("name").text:
```

Comment:
I like the walrus operator. However, It's only supporter for python > 3.8 so I would rather not use it here.
```suggestion
            channel_name = channel.find("name").text
            if channel_name:
```

Reply:
yes, this i a good point.